pool.go

```diff
@@ -3,22 +3,37 @@ package pool
 
 import (
 	"errors"
+	"time"
 )
 
 var (
 	// ErrClosed is the error resulting if the pool is closed via pool.Close().
-	ErrClosed = errors.New("pool is closed")
+	ErrClosed   = errors.New("pool is closed")
+	ErrTimedOut = errors.New("timed out waiting for connection")
 )
 
+type GenericConn interface{}
+
+type ConnectionHolder struct {
+	Conn  GenericConn
+	InUse bool
```

Comment:
I'm not really sure what the purpose of the InUse flag is. Is it just to cover some edge cases in closing down the pool?